token/typed.go

```diff
@@ -0,0 +1,59 @@
+package token
+
+import (
+	"fmt"
+	"sort"
+)
+
+// ArgumentsParser defines a parser for the arguments of a typed token.
+// Parsing stops unrecoverably at the first error. The return value of Err must be checked before using the values returned by precedings calls to GetInt.
+type ArgumentsParser interface {
+	// GetInt tries to parse the argument name and returns its integer value or defaultValue it the argument is not found.
+	// The return value is valid only if Err returns nil.
+	GetInt(name string, defaultValue int) int
+	// Err returns the first error encountered by the ArgumentsParser.
+	Err() error
+}
+
+// CreateTyped defines a function to create a typed token
+type CreateTyped func(argParser ArgumentsParser) (Token, error)
```

Comment:
Well, it is recommended not to be redundant in the naming of a package and the functions it contains in Go. That is why I was suggesting to move this to a separate `token/typed` package. Functions would look like `typed.List`, `typed.Register`, and the global variable would have a more limited scope.